lib/matplotlib/figure.py

```diff
@@ -1597,6 +1597,175 @@ def subplots(self, nrows=1, ncols=1, sharex=False, sharey=False,
             # Returned axis array will be always 2-d, even if nrows=ncols=1.
             return axarr
 
+    @staticmethod
+    def _normalize_grid_string(layout):
+        return [list(ln) for ln in layout.strip('\n').split('\n')]
+
+    def build_grid(self, layout, *, subplot_kw=None, gridspec_kw=None,
+                   empty_sentinel=' '):
+        """
+        Build a layout of Axes based on 2D layout.
+
+        This is a helper function to build complex GridSpec layouts visually.
+
+        Parameters
+        ----------
+        layout : list of list of {hashable or nested} or str
+
+            A visual layout of how you want your Axes to be arranged
+            labeled as strings.  For example ::
+
+               x = [['A', 'A', 'edge'],
+                    ['C', 'D', 'edge']]
+
+            Produces 4 axes:
+
+            - 'A' which is 1 row high and spans the first two columns
+            - 'edge' which is 2 rows high and is on the right edge
+            - 'C' which in 1 row and 1 column wide in the bottom left
+            - 'D' which in 1 row and 1 column wide in the bottom center
+
+            If input is a str, then it must be of the form ::
+
+              '''
+              AAB
+              CDD
+              '''
+
+            where the labels are assumed to be single characters and
+            the rows are delimited by new lines.  This
+
+        subplot_kw : dict, optional
+            Dict with keywords passed to the
+            :meth:`~matplotlib.figure.Figure.add_subplot` call used to create
+            each subplot.
+
+        gridspec_kw : dict, optional
+            Dict with keywords passed to the
+            `~matplotlib.gridspec.GridSpec` constructor used to create
+            the grid the subplots are placed on.
+
+        empty_sentinel : object, optional
+            Entry in the layout to mean "leave this space empty".  Defaults
+            to ``' '`` (single space).
+
+        Returns
+        -------
+        dict[str, Axes]
+           A dictionary mapping the string labels to the new Axes objects.
+
+        """
+        subplot_kw = subplot_kw or {}
+        gridspec_kw = gridspec_kw or {}
+        # special-case string input
+        if isinstance(layout, str):
+            layout = self._normalize_grid_string(layout)
+
+        def _make_array(inp):
+            """Convert input into 2D array
```

Comment:
flake8-docstrings is confused about line numbers, but this is the one that flake8 is posting about. There should be a newline after `"""`.